fe/src/components/ViewToggler/ViewToggler.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import {
+  Stack,
+  Button,
+  Icon,
+} from '@chakra-ui/react';
+import { BsListUl, BsFillGrid3X3GapFill } from 'react-icons/bs';
+import { HomeContext } from '../../pages/Home';
+import { ViewType } from '../../utils';
+
+interface Props {
+  viewSwitchable: boolean;
+}
+
+const ViewToggler: React.FC<Props> = ({ viewSwitchable } : Props): JSX.Element => {
+  const { view, setView } = React.useContext(HomeContext);
+  const isView = (viewType: ViewType): boolean => view === viewType;
+
+  const setViewType = (): void => {
+    if (!viewSwitchable) {
+      return;
+    }
+
+    const newViewType: ViewType = isView(ViewType.List)
+      ? ViewType.Grid
+      : ViewType.List;
+
+    setView(newViewType);
+  };
+
+  React.useEffect(() => {
+    if (!viewSwitchable && (view !== ViewType.Grid)) {
+      setView(ViewType.Grid);
+    }
+  }, [view, setView, viewSwitchable]);
```

Comment:
In case you're wondering why i updated this to `useEffect`. I ran into this error and found that this was a possible fix for that -> https://flaviocopes.com/react-update-while-rendering-different-component/

![error](https://user-images.githubusercontent.com/48998862/117823269-f5d18980-b275-11eb-84d6-4f3c187c211c.JPG)